modules/dem_uart/test/test_dem_uart.py

```diff
@@ -0,0 +1,222 @@
+                                     mod_vendor=1, mod_type=5, mod_version=0,
+                                     can_stall=True)
+
+@cocotb.coroutine
+def _activate_module(dut):
+    """
+    Set the MOD_CS_ACTIVE bit in the Control and Status register to 1 to
+    enable transmission of debug event packets.
+    """
+
+    access = RegAccess(dut)
+
+    yield access.write_register(dest=MODULE_DI_ADDRESS,
+                                src=SENDER_DI_ADDRESS,
+                                word_width=16,
+                                regaddr=DiPacket.BASE_REG.MOD_CS.value,
+                                value=1)
+
+@cocotb.test()
+def test_dem_uart_activation(dut):
+    """
+    Check if DEM_UART is handling the activation bit correctly
+    """
+
+    access = RegAccess(dut)
+
+    yield _init_dut(dut)
+
+    dut._log.info("Check contents of MOD_CS")
+    yield access.assert_reg_value(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
+                                  DiPacket.BASE_REG.MOD_CS.value, 16, 0)
+
+    yield _activate_module(dut)
+
+    dut._log.info("Check contents of MOD_CS")
+    yield access.assert_reg_value(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
+                                  DiPacket.BASE_REG.MOD_CS.value, 16, 1)
+
+# Mirroring expected transfer through module
+_fifo = []
+
+@cocotb.test()
+def test_dem_to_dii(dut):
+    """
+    Verify that no chars are lost when the DII is busy
+    """
+
+    yield _init_dut(dut)
+    yield _activate_module(dut)
+
+    write_thread = cocotb.fork(dem_to_dii_tx(dut, num_transfers=1000, random_data=True))
+    read_thread  = cocotb.fork(dem_to_dii_rx(dut, num_transfers=1000))
+    
+    # read_thread implicitly waits for write thread to finish as well
+    yield read_thread.join()
+
+@cocotb.coroutine
+def dem_to_dii_tx(dut, num_transfers=500, max_delay=5, random_data=False):
+
+    for i in range(num_transfers):
+
+        delay = random.randint(0, max_delay)
+
+        for _ in range(delay):
+            yield RisingEdge(dut.clk)
+
+        data = (random.randint(0, 255) if random_data else 0x42)
+
+        _fifo.append(data)
+
+        dut.out_valid <= 0x1
+        dut.out_char <= int(data)
+
+        # out_char & out_valid must be set until both out_ready & out_valid
+        # are asserted at the same rising edge, and for at least one clock cycle 
+        yield RisingEdge(dut.clk)
+
+        while not dut.out_ready.value or not dut.out_valid:
+            yield RisingEdge(dut.clk)
+
+        # Negate out_valid after the module has read out_char
+        dut.out_valid <= 0
+
+@cocotb.coroutine
+def dem_to_dii_rx(dut, num_transfers=500, max_delay=5):
+
+    reader    = NocDiReader(dut, dut.clk);
+    ex_packet = DiPacket();
+
+    for i in range(num_transfers):
+
+        delay = random.randint(1, max(1, max_delay))
+
+        for _ in range(delay):
+            yield RisingEdge(dut.clk)
+
+        rxPacket = yield reader.receive_packet(set_ready=True)
```

Comment:
The coding style in this file are identifiers with underscores, not camelCase ones: `rx_packet`